src/index.js

```diff
@@ -23,10 +29,49 @@ const products = [
   { name: 'laptop', category: 'computer' },
 ]
 // Target the ul element in index.html and render the above list
+let list = document.querySelector("ul");
 
+function addList(){
+  for(let i = 0; i < products.length; i++){
+    let newTodo = document.createElement("li");
+    newTodo.innerHTML = products[i].name;
+    //console.log(newTodo);
+    list.appendChild(newTodo);
+  }
+}
+addList();
+    
 // 7. Target the button in index.html, attach an event listener to it. Every time we click
 // on the button, it will toggle the product list. Show => hide, hide => show
+let button  = document.querySelector("button");
+function toggle() {
+  if(list.style.display === "none"){
+    list.style.display = "block";
+  }else {
+    list.style.display = "none"
+  }
+  
+} 
+button.addEventListener("click", toggle);
+
 
 // 8. Target the input element and attach an event listener to it.
 // Once we click search button, you need to use the text input to search for the relevant product name from the list above
 // and show it in the <span> tag
+let input = document.querySelector("input");
+let search = document.querySelector("#search");
+let result = document.querySelector(".result");
+function results(e) {
+  e.preventDefault();
+  let word = input.value;
+  console.log(word);
+  const searchResult = products.filter(product => {
+    //console.log(product.name.slice(0, word.length), word);
+   if(product.name.slice(0, word.length) === word){
```

Comment:
The easier way is to check if the product name includes the word as a sub string. Check string.include()

Reply:
if(product.name.includes(word)){
     return result.innerHTML = product.name;